com.hazelcast.jet.core.TopologyChangeTest.when_nodeIsNotJobParticipant_then_initFails is failing with "Mismatch between coordinator and participant version" error

master, rev. d372406

Failed on compatibility-jdk, Oracle JDK 8.

http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/853/testReport/
Stacktrace:
```
java.lang.AssertionError: com.hazelcast.jet.JetException: Mismatch between coordinator and participant version is not an instanceof java.lang.IllegalArgumentException
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.hazelcast.test.HazelcastTestSupport.assertInstanceOf(HazelcastTestSupport.java:930)
	at com.hazelcast.jet.core.TopologyChangeTest.when_nodeIsNotJobParticipant_then_initFails(TopologyChangeTest.java:525)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Finished Running Test: when_jobParticipantReceivesStaleInitOperation_then_jobRestarts[liteMemberFlags(0)] in 3.837 seconds.
Started Running Test: when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]
02:18:02,024  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:18:02,025  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:18:02,025  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
02:18:02,025  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210714 - d372406) starting at [127.0.0.1]:5701
02:18:02,025  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Cluster name: dev
02:18:02,028  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:18:02,032  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:18:02,035  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 4
02:18:02,035  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:18:02,035  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTING
02:18:02,036  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea this
]

02:18:02,036  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Jet is enabled
02:18:02,036  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTED
02:18:02,036 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,036  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:18:02,037  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [logo] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:18:02,037  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
02:18:02,037  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210714 - d372406) starting at [127.0.0.1]:5702
02:18:02,037  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Cluster name: dev
02:18:02,040  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:18:02,043  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [CPSubsystem] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:18:02,046  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetServiceBackend] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 4
02:18:02,047  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Diagnostics] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:18:02,047  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5702 is STARTING
02:18:02,047  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=true}
02:18:02,047  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] hz.loving_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=true}
02:18:02,047  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.loving_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea this
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4
]

02:18:02,136 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,148  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.frosty_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4 this
]

02:18:02,236 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,336 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,436 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,536 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,547  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetExtension] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Jet is enabled
02:18:02,547  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5702 is STARTED
02:18:02,548  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:18:02,549  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [logo] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:18:02,549  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
02:18:02,549  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210714 - d372406) starting at [127.0.0.1]:5703
02:18:02,549  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [system] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Cluster name: dev
02:18:02,552  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:18:02,557  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [CPSubsystem] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:18:02,559  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetServiceBackend] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 4
02:18:02,560  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Diagnostics] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:18:02,560  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is STARTING
02:18:02,561  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5701, alive=true}
02:18:02,561  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] hz.loving_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5703, alive=true}
02:18:02,561  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.loving_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea this
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4
	Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
]

02:18:02,561  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] hz.frosty_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5703, alive=true}
02:18:02,561  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.frosty_heyrovsky.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4 this
	Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
]

02:18:02,637 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,661  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.stoic_heyrovsky.generic-operation.thread-1 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4
	Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef this
]

02:18:02,737 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,837 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:02,937 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:03,037 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:18:03,061  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5702, alive=true}
02:18:03,061  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetExtension] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Jet is enabled
02:18:03,061  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Time-limited test - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is STARTED
02:18:03,063  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [PartitionStateManager] when_nodeIsNotJobParticipant_then_initFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Initializing cluster partition table arrangement...
02:18:03,064 ERROR |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [InitExecutionOperation] when_nodeIsNotJobParticipant_then_initFails - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Mismatch between coordinator and participant version
com.hazelcast.jet.JetException: Mismatch between coordinator and participant version
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.doRun(InitExecutionOperation.java:86) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:53) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:272) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:600) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:579) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:240) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) [classes/:?]
	at com.hazelcast.jet.core.TopologyChangeTest.when_nodeIsNotJobParticipant_then_initFails(TopologyChangeTest.java:519) [test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_251]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_251]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_251]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_251]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115) [test-classes/:?]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107) [test-classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_251]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]
02:18:03,324  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Starting job 0000-0000-0000-0001: discovered by scanning of JobRecords
02:18:03,324 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobRepository] hz.loving_heyrovsky.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Job cleanup took 0ms
02:18:03,325 ERROR |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MasterJobContext] hz.loving_heyrovsky.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Execution of job '0000-0000-0000-0001', execution 0000-0000-0000-0000 failed
	Start time: 2021-07-14T02:18:03.324
	Duration: 00:00:00.001
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
java.lang.NullPointerException: null
	at com.hazelcast.jet.impl.MasterJobContext.resolveDagAndCL(MasterJobContext.java:300) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:201) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$52(JobCoordinationService.java:1222) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$53(JobCoordinationService.java:1243) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [classes/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_251]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_251]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
02:18:03,326 DEBUG |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JobCoordinationService] hz.loving_heyrovsky.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] job '0000-0000-0000-0001', execution 0000-0000-0000-0000 is completed
02:18:03,328  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [JetTestSupport] Time-limited test - Terminating instanceFactory in JetTestSupport.@After
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is SHUTTING_DOWN
02:18:03,329  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Terminating forcefully...
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Shutting down connection manager...
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5703, alive=false}
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5702, alive=false}
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5703, alive=false}
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5701, alive=false}
02:18:03,329  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MembershipManager] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef is suspected to be dead for reason: Connection manager is stopped on Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef this
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MembershipManager] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Removing Member [127.0.0.1]:5703 - 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [PartitionStateManager] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [PartitionStateManager] hz.frosty_heyrovsky.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea this
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4
]

02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [TransactionManagerService] hz.loving_heyrovsky.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
02:18:03,329  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Shutting down node engine...
02:18:03,330  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [TransactionManagerService] hz.frosty_heyrovsky.cached.thread-3 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 4218f0e6-b1e6-4dfd-adfb-04c9d94661ef
02:18:03,330  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] hz.frosty_heyrovsky.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea
	Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4 this
]

02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [NodeExtension] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Destroying node NodeExtension.
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5703 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5703 is SHUTDOWN
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
02:18:03,331  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Terminating forcefully...
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Shutting down connection manager...
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=false}
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MockServer] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=false}
02:18:03,331  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MembershipManager] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 761b13bf-63a9-4926-a314-d98126cca9f4
02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MigrationManager] hz.loving_heyrovsky.migration - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Repartitioning cluster data. Migration tasks count: 6
02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [PartitionStateManager] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 761b13bf-63a9-4926-a314-d98126cca9f4
02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [ClusterService] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:1, ver:5} [
	Member [127.0.0.1]:5701 - d57ee94a-61d5-48fd-a8d3-a6bab90ef4ea this
]

02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [TransactionManagerService] hz.loving_heyrovsky.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 761b13bf-63a9-4926-a314-d98126cca9f4
02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Shutting down node engine...
02:18:03,332  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [MigrationManager] hz.loving_heyrovsky.migration - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Rebalance process was aborted. Ignoring remaining migrations. Will recalculate the new migration plan. (repartitionTime=Wed Jul 14 02:18:03 UTC 2021, plannedMigrations=6, completedMigrations=0, remainingMigrations=6, totalCompletedMigrations=0, elapsedMigrationOperationTime=0ms, totalElapsedMigrationOperationTime=0ms, elapsedDestinationCommitTime=0ms, totalElapsedDestinationCommitTime=0ms, elapsedMigrationTime=0ms, totalElapsedMigrationTime=0ms)
02:18:03,333  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [NodeExtension] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Destroying node NodeExtension.
02:18:03,334  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
02:18:03,334  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
02:18:03,334  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
02:18:03,334  WARN |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Terminating forcefully...
02:18:03,334  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Shutting down connection manager...
02:18:03,334  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Shutting down node engine...
02:18:03,335  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [NodeExtension] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Destroying node NodeExtension.
02:18:03,335  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [Node] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 1 ms.
02:18:03,335  INFO |when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)]| - [LifecycleService] Thread-4981 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)](com.hazelcast.jet.core.TopologyChangeTest): BuildInfo{version='5.0-SNAPSHOT', build='20210714', buildNumber=20210714, revision=d372406, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'when_nodeIsNotJobParticipant_then_initFails[liteMemberFlags(0)](com.hazelcast.jet.core.TopologyChangeTest):'
02:18:00, accumulated pauses: 276 ms, max pause: 255 ms, pauses over 1000 ms: 0


No metrics recorded during the test```

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/TopologyChangeTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/TopologyChangeTest.java
@@ -513,7 +513,7 @@ public class TopologyChangeTest extends JetTestSupport {
         JobRecord jobRecord = new JobRecord(version, jobId, null, "", new JobConfig(), Collections.emptySet(), null);
         instances[0].getMap(JOB_RECORDS_MAP_NAME).put(jobId, jobRecord);
 
-        InitExecutionOperation op = new InitExecutionOperation(jobId, executionId, memberListVersion, null, memberInfos, null, false);
+        InitExecutionOperation op = new InitExecutionOperation(jobId, executionId, memberListVersion, version, memberInfos, null, false);
         Future<Object> future = Accessors.getOperationService(master)
                 .createInvocationBuilder(JetServiceBackend.SERVICE_NAME, op, Accessors.getAddress(master))
                 .invoke();

